### PR TITLE
cliquer: init at 1.21

### PR DESCRIPTION
**Pull request by timokau**
###### Motivation for this change

Package cliquer.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/science/math/cliquer/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+  buildInputs = [
+    autoreconfHook
+  ];
+
+  meta = with stdenv.lib; {
+    homepage = https://users.aalto.fi/~pat/cliquer.html;
+    description = "Routines for clique searching";
+    longDescription = ''
+      Cliquer is a set of C routines for finding cliques in an arbitrary weighted graph. It uses an exact branch-and-bound algorithm developed by Patric Östergård. It is designed with the aim of being efficient while still being flexible and easy to use.
```
Comment (dotlambda) [inline pkgs/development/libraries/science/math/cliquer/default.nix]: line breaks

File: pkgs/development/libraries/science/math/cliquer/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    '';
+    license = with licenses; [ gpl2 ];
+    maintainers = with maintainers; [ timokau ];
+    platforms = platforms.all;
```
Comment (dotlambda) [inline pkgs/development/libraries/science/math/cliquer/default.nix]: Tested?

File: pkgs/development/libraries/science/math/cliquer/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{ stdenv
+, fetchFromGitHub
+, autoreconfHook
+}:
+
+stdenv.mkDerivation rec {
+  version = "1.21";
+  name = "cliquer-${version}";
+
+  # autotoolized version of the original cliquer
```
Comment (dotlambda) [inline pkgs/development/libraries/science/math/cliquer/default.nix]: I don't think this package should have the name cliquer then. And the homepage should also be changed accordingly.

Comment (timokau) [reply pkgs/development/libraries/science/math/cliquer/default.nix]: Its still the same package with the same functionality, just an improved / updated build. I think thats an implementation detail that would only confuse users that want to install cliquer.

However if you prefer a name change and have a different name in mind, I'm not strongly opposed.

Comment (dotlambda) [reply pkgs/development/libraries/science/math/cliquer/default.nix]: Considering that Arch is also using autocliquer, this seems alright after all: https://git.archlinux.org/svntogit/community.git/tree/trunk/PKGBUILD?h=packages/cliquer#n15
However please add a `downloadPage` pointing to autocliquer.

Comment (dotlambda) [reply pkgs/development/libraries/science/math/cliquer/default.nix]: You could use
```
downloadPage = src.meta.homepage;
```

File: pkgs/development/libraries/science/math/cliquer/default.nix [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    homepage = https://users.aalto.fi/~pat/cliquer.html;
+    downloadPage = src.meta.homepage; # autocliquer
+    description = "Routines for clique searching";
+    longDescription = ''
+      Cliquer is a set of C routines for finding cliques in an arbitrary weighted graph.
+      It uses an exact branch-and-bound algorithm developed by Patric Östergård.
+      It is designed with the aim of being efficient while still being flexible and
+      easy to use.
+    '';
+    license = with licenses; [ gpl2 ];
```
Comment (dotlambda) [inline pkgs/development/libraries/science/math/cliquer/default.nix]: Please get rid of the list and just use `licenses.gpl2Plus`:

> Cliquer is licensed under the GNU General Public License as published
by the Free Software Foundation; either version 2 of the License, or
(at your option) any later version.

(from the README in the tarball)

**Pull request merged**